apps/minifront/src/components/shared/non-native-fee-warning.tsx

```diff
@@ -1,38 +1,105 @@
   balancesResponses: BalancesResponse[] = [],
   stakingAssetMetadata?: Metadata,
 ): boolean => {
-  return balancesResponses.some(asset =>
-    getAssetIdFromValueView
-      .optional()(asset.balanceView)
-      ?.equals(getAssetId.optional()(stakingAssetMetadata)),
+  return balancesResponses.some(
+    asset =>
+      getAssetIdFromValueView
+        .optional()(asset.balanceView)
+        ?.equals(getAssetId.optional()(stakingAssetMetadata)) &&
+      getAddressIndex.optional()(asset)?.account === account,
   );
 };
 
-export const useShouldRender = (balancesResponses: BalancesResponse[] = [], amount: number) => {
+// Finds the alt tokens in the user's account balances that can be used for fees
+const hasAltToken = (
+  account: number,
+  balancesResponses: BalancesResponse[] = [],
+  gasPrices: GasPrices[],
+) => {
+  const accountAssets = balancesResponses.filter(
+    balance => getAddressIndex.optional()(balance)?.account === account,
+  );
+  return accountAssets.some(balance => {
+    const amount = getAmount(balance);
+    const hasBalance = amount.lo !== 0n || amount.hi !== 0n;
+    if (!hasBalance) {
+      return false;
+    }
+
+    return gasPrices.some(price =>
+      price.assetId?.equals(getAssetIdFromBalancesResponseOptional(balance)),
+    );
+  });
+};
+
+const useGasPrices = () => {
+  const [prices, setPrices] = useState<GasPrices[]>([]);
+
+  const fetchGasPrices = useCallback(async () => {
+    const res = await viewClient.gasPrices({});
+    setPrices(res.altGasPrices);
+  }, []);
+
+  useEffect(() => {
+    void fetchGasPrices();
+  }, [fetchGasPrices]);
+
+  return prices;
+};
+
+// Returns boolean if the Alt fees will be used for transaction
+export const useShouldRender = (
+  balancesResponses: BalancesResponse[] = [],
+  amount: number,
+  account?: BalancesResponse,
+) => {
+  const gasPrices = useGasPrices();
   const stakingTokenMetadata = useStakingTokenMetadata();
-  const userHasStakingToken = hasStakingToken(balancesResponses, stakingTokenMetadata.data);
-  const showNonNativeFeeWarning = amount > 0 && !userHasStakingToken;
 
-  return showNonNativeFeeWarning;
+  if (!amount) {
+    return false;
+  }
+
+  const sourceAddressIndex = getAddressIndex.optional()(account)?.account ?? 0;
+  const userHasStakingToken = hasStakingToken(
+    sourceAddressIndex,
+    balancesResponses,
+    stakingTokenMetadata.data,
+  );
+
+  if (userHasStakingToken) {
+    return false;
+  }
+
+  return hasAltToken(sourceAddressIndex, balancesResponses, gasPrices);
 };
 
 /**
- * Renders a non-native fee warning if A) the user does not have any balance of
- * the staking token to use for fees, and B) the amount the user has entered for
- * a transaction (e.g., send or swap) is nonzero -- i.e., a fee will be
- * required.
+ * Renders a non-native fee warning if
+ * 1. the user does not have any balance (in the selected account) of the staking token to use for fees
+ * 2. user has alt token balances to pay for the fees
```

Comment:
"The user does not have sufficient balances in alternative tokens to cover the fees."